### PR TITLE
RequeueStmt nameres & separate navigation

### PR DESCRIPTION
**Pull request by raph-amiard**

### CODE REVIEW

File: ada/language/ast.py [hunk 1]
```diff
@@ -9572,6 +9579,14 @@ class EntryDecl(BasicDecl):
 
     defining_names = Property(Entity.spec.name.as_entity.singleton)
 
+    @langkit_property()
+    def match_params(params=T.BaseFormalParamHolder.entity):
+        """
+        Wether Self matches the params denoted by ``params``. Used as a logic
```
Comment (pmderodat) [inline ada/language/ast.py]: Typo: `whether` (missing H)

File: ada/language/ast.py [hunk 2]
```diff
@@ -2268,6 +2268,29 @@ def real_type(tpe=T.BaseTypeDecl.entity):
             tpe
         )
 
+    @langkit_property(return_type=Bool)
+    def match_formal_params(other=T.BaseFormalParamHolder.entity,
+                            match_names=(Bool, True)):
+        """
+        Check wether self's params match other's.
```
Comment (pmderodat) [inline ada/language/ast.py]: Typo (whether)

Comment (pmderodat) [reply ada/language/ast.py]: (nevertheless, many thanks for adding the docstring)

File: ada/language/ast.py [hunk 3]
```diff
@@ -2000,6 +2000,22 @@ def unbound_previous_part():
                 Self, Entity.defining_name.all_env_els_impl
             ).at(0).cast(T.BasicDecl),
             lambda _: No(T.BasicDecl.entity),
+        ))
+
+        # HACK: All previous_part implems except the one for subprograms skip
```
Comment (Roldak) [inline ada/language/ast.py]: Omg that is horrible.
Approved.

File: ada/testsuite/tests/name_resolution/requeue_stmt/test_requeue.adb [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+
+   protected body Producer_Consumer is
+
+      entry Get_Data (Value_Get : out Natural) when true is
+      begin
+         if Data_Available then
+            Value_Get := The_Data;
+            Data_Available := False;
+         else
+            requeue Wait_For_Get;
```
Comment (Roldak) [inline ada/testsuite/tests/name_resolution/requeue_stmt/test_requeue.adb]: Looking at the nameres logic you added, I think this is missing a few test cases!

**Pull request merged**